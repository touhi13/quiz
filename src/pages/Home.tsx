import { Box, Button, Container, Grid, MenuItem, TextField, Typography } from '@mui/material';
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';

export interface ExamineeDetails {
    name: string,
    gender: string,
    lang: string
}

const Home: React.FC = () => {
    const navigate = useNavigate()
    const [examinee, setExamineeDetails] = useState<ExamineeDetails>({ name: "", gender: "Male", lang: "JavaScript" })

    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = e.target;
        setExamineeDetails({ ...examinee, [name]: value });
    }
    const handleSubmit = () => {
        console.log(examinee);
        navigate('/exam', { state: examinee })
    }
    return (

        <Container component="main" maxWidth="xs">
            <Box sx={{
                marginTop: 8,
            }}>
                <Grid container spacing={2}>
                    <Grid item xs={12}>
                        <Typography variant='h4'>User Information</Typography>
                    </Grid>
                    <Grid item xs={12} >
                        <TextField inputProps={{ "data-testid": "addName" }}
                            fullWidth name="name" value={examinee.name} onChange={handleInputChange} ></TextField>
                    </Grid>

                    <Grid item xs={12} >
                        <TextField fullWidth name="gender" value={examinee.gender} onChange={handleInputChange} select>
                            <MenuItem value="Male">Male</MenuItem>
                            <MenuItem value="Female">Female</MenuItem>
                        </TextField>
                    </Grid>
                    <Grid item xs={12} >
                        <TextField fullWidth name="lang" value={examinee.lang} onChange={handleInputChange} select>
                            <MenuItem value="JavaScript">JavaScript</MenuItem>
                            <MenuItem value="PHP">PHP</MenuItem>
                        </TextField>
                    </Grid>
                    <Grid item xs={12}>
                        <Button fullWidth variant="contained" data-testid="submitButton" disabled={examinee.name.length < 1} onClick={handleSubmit}>Submit</Button>
                    </Grid>
                </Grid>
            </Box>
        </Container>

    );
};

export default Home;