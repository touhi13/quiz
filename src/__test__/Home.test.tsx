import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Home from '../pages/Home';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { act as domAct } from "react-dom/test-utils";
import { act as testAct, create } from "react-test-renderer";


describe('Home component', () => {
    it('should render the component onto the screen', () => {
        expect(true).toBeTruthy();
    });

    it('should have the "Submit" button disabled when initialized', () => {

        render(
            <BrowserRouter>
                <Routes>
                    <Route path="/" element={<Home />} />
                </Routes>
            </BrowserRouter>
        );
        expect(screen.getByTestId('submitButton')).toBeDisabled();
    });

    it('should enable the "Submit" button when a valid input is entered', () => {
        render(
            <BrowserRouter>
                <Routes>
                    <Route path="/" element={<Home />} />
                </Routes>
            </BrowserRouter>
        );
        expect(screen.getByTestId('submitButton')).toBeDisabled();
        const input = screen.getByTestId('addName');
        fireEvent.change(input, { target: { value: 'John' } });
        expect(screen.getByTestId('submitButton')).toBeEnabled();
    });
    let root;
    domAct(() => {
        testAct(() => {
            root = create(
                <BrowserRouter>
                    <Routes>
                        <Route path="/" element={<Home />} />
                    </Routes>
                </BrowserRouter>);
        });
    });
    expect(root).toMatchSnapshot();
});


// ...
