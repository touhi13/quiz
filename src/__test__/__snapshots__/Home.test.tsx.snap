// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
>
  <div
    className="MuiBox-root css-binzgt"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          User Information
        </h4>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="addName"
              disabled={false}
              id="mui-1"
              name="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-nnbavb"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
            variant="outlined"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="mui-2"
              className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-2"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Male
            </div>
            <input
              aria-hidden={true}
              autoFocus={false}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              name="gender"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="Male"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-nnbavb"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
            variant="outlined"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="mui-3"
              className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-3"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              React
            </div>
            <input
              aria-hidden={true}
              autoFocus={false}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              name="lang"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value="React"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-nnbavb"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          data-testid="submitButton"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</main>
`;
